packages/office-ui-fabric-react/src/components/ContextualMenu/ContextualMenu.base.tsx

```diff
@@ -382,20 +389,40 @@ export class ContextualMenuBase extends BaseComponent<IContextualMenuProps, ICon
     // We only send a dividerClassName when the item to be rendered is a divider. For all other cases, the default divider style is used.
     const dividerClassName = item.itemType === ContextualMenuItemType.Divider ? item.className : undefined;
     const subMenuIconClassName = item.submenuIconProps ? item.submenuIconProps.className : '';
-    const getTheItemClassNames = item.getItemClassNames || getItemClassNames;
-    const itemClassNames = getTheItemClassNames(
-      this.props.theme!,
-      isItemDisabled(item),
-      this.state.expandedMenuItemKey === item.key,
-      !!getIsChecked(item),
-      !!item.href,
-      iconProps.iconName !== 'None',
-      item.className,
-      dividerClassName,
-      iconProps.className,
-      subMenuIconClassName,
-      item.primaryDisabled
-    );
+
+    let itemClassNames;
+
+    // IContextualMenuItem#getItemClassNames for backwards compatibility
+    // otherwise uses mergeStyles for class names.
+    if (item.getItemClassNames) {
+      itemClassNames = item.getItemClassNames(
+        this.props.theme!,
+        isItemDisabled(item),
+        this.state.expandedMenuItemKey === item.key,
+        !!getIsChecked(item),
+        !!item.href,
+        iconProps.iconName !== 'None',
+        item.className,
+        dividerClassName,
+        iconProps.className,
+        subMenuIconClassName,
+        item.primaryDisabled
+      );
+    } else {
+      itemClassNames = getItemClassNames(getItemStyles, {
```

Comment:
We could merge the output of both if both defined... not sure if there's a precedence for this.